### PR TITLE
Feature: sigmoid functions, makeDiagonal

### PR DESCRIPTION
**Pull request by OlegStotsky**

### CODE REVIEW

File: src/lib/utils/MathExtra.ts [hunk 1]
```diff
@@ -327,6 +328,69 @@ const inner = (a, b) => {
...
+ * Computes sigmoid function in point x
+ * @param x - point in which sigmoid is computed
+ */
+const sigmoid = (x: number): number => {
+  if (x >= 0) {
+    const z = Math.exp(-x);
+    return 1 / (1 + z);
+  } else {
+    const z = Math.exp(x);
+    return z / (1 + z);
```
Comment (JasonShin) [inline src/lib/utils/MathExtra.ts]: why can't we just put all these in one line? 

`Math.exp(x) / (1 + z)`

Comment (OlegStotsky) [reply src/lib/utils/MathExtra.ts]: Because that way we save one operation.

File: src/lib/utils/MathExtra.ts [hunk 2]
```diff
@@ -327,6 +328,69 @@ const inner = (a, b) => {
   throw new Error(`Cannot process with the invalid inputs ${a} and ${b}`);
 };
 
+/**
+ * Converts 1D matrix of size N to a 2D diagonal matrix of size NxN
+ * @example makeDiagonal([1, 2, 3])
+ * returns [[1, 0, 0], [0, 2, 0], [0, 0, 3]
+ * @param x - source array
+ */
+const makeDiagonal = (X: Type1DMatrix<number>): Type2DMatrix<number> => {
```
Comment (JasonShin) [inline src/lib/utils/MathExtra.ts]: numericjs has diag, can't we just use that?

File: src/lib/utils/MathExtra.ts (hunk 1 above)
Comment (JasonShin) [reply src/lib/utils/MathExtra.ts]: ah

**Pull request closed without merging**